File: hw_6_async/src/main/java/com/gmail/elnora/fet/hw_6_async/data/DataTypeConverter.java
package com.gmail.elnora.fet.hw_6_async.data;

import androidx.room.TypeConverter;

public class DataTypeConverter {
    @TypeConverter
    public static String fromDataTypeToString(DataType dataType) {
        if (dataType == null)
            return null;
        return dataType.toString();
    }

    @TypeConverter
    public static DataType fromStringToDataType(String dataType) {
        if (dataType.equals(DataType.EMAIL.toString())) {
            return DataType.EMAIL;
        } else if (dataType.equals(DataType.PHONE.toString())) {
            return DataType.PHONE;
        } else {
            throw new IllegalArgumentException("Couldn't recognize dataType");
        }
    }
}

File: hw_5_contacts_database_kotlin/src/main/java/com/gmail/elnora/fet/hw_5_contacts_database_kotlin/DataTypeConverter.kt
package com.gmail.elnora.fet.hw_5_contacts_database_kotlin

import androidx.room.TypeConverter

object DataTypeConverter {
    @JvmStatic
    @TypeConverter
    fun fromDataTypeToString(dataType: DataType): String = dataType.toString()

    @JvmStatic
    @TypeConverter
    fun fromStringToDataType(dataType: String): DataType {
        return when (dataType) {
            DataType.EMAIL.toString() -> {
                DataType.EMAIL
            }
            DataType.PHONE.toString() -> {
                DataType.PHONE
            }
            else -> {
                throw IllegalArgumentException("Couldn't recognize dataType")
            }
        }
    }
}